FirebaseAdmin/FirebaseAdmin/Auth/Providers/ListProviderConfigsRequest.cs

```diff
@@ -0,0 +1,165 @@
+
+        private readonly string baseUrl;
+        private readonly ErrorHandlingHttpClient<FirebaseAuthException> httpClient;
+
+        protected ListProviderConfigsRequest(
+            string baseUrl,
+            ErrorHandlingHttpClient<FirebaseAuthException> httpClient,
+            ListProviderConfigsOptions options)
+        {
+            this.baseUrl = baseUrl;
+            this.httpClient = httpClient;
+            this.RequestParameters = new Dictionary<string, IParameter>();
+            this.SetPageSize(options?.PageSize);
+            this.SetPageToken(options?.PageToken);
+        }
+
+        public abstract string MethodName { get; }
+
+        public abstract string RestPath { get; }
+
+        public string HttpMethod => "GET";
+
+        public IDictionary<string, IParameter> RequestParameters { get; }
+
+        public IClientService Service { get; }
+
+        public HttpRequestMessage CreateRequest(bool? overrideGZipEnabled = null)
+        {
+            var queryParameters = string.Join("&", this.RequestParameters.Select(
+                kvp => $"{kvp.Key}={kvp.Value.DefaultValue}"));
+            var request = new HttpRequestMessage()
+            {
+                Method = System.Net.Http.HttpMethod.Get,
+                RequestUri = new Uri($"{this.baseUrl}/{this.RestPath}?{queryParameters}"),
+            };
+            request.Headers.Add(
+                ProviderConfigManager.ClientVersionHeader, ProviderConfigManager.ClientVersion);
+            return request;
+        }
+
+        public async Task<Stream> ExecuteAsStreamAsync(CancellationToken cancellationToken)
+        {
+            var response = await this.httpClient.SendAsync(this.CreateRequest(), cancellationToken)
+                .ConfigureAwait(false);
+            return await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        }
+
+        public Stream ExecuteAsStream()
+        {
+            return this.ExecuteAsStreamAsync().Result;
+        }
+
+        public async Task<Stream> ExecuteAsStreamAsync()
+        {
+            return await this.ExecuteAsStreamAsync(default).ConfigureAwait(false);
+        }
+
+        public AuthProviderConfigs<T> Execute()
+        {
+            return this.ExecuteAsync().Result;
+        }
+
+        public async Task<AuthProviderConfigs<T>> ExecuteAsync()
+        {
+            return await this.ExecuteAsync(default).ConfigureAwait(false);
+        }
+
+        public async Task<AuthProviderConfigs<T>> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            var jObject = await this.SendAndDeserializeAsync(cancellationToken)
+                .ConfigureAwait(false);
+            return this.BuildProviderConfigs(jObject);
+        }
+
+        internal void SetPageSize(int? pageSize)
+        {
+            if (pageSize > MaxListResults)
+            {
+                throw new ArgumentException("Page size must not exceed 100.");
```

Comment:
nit: might be nice to interpolate `MaxListResults` here instead of hardcoding 100

Reply:
Good call. Done.